sockeye/image_captioning/captioner.py

```diff
@@ -127,6 +127,13 @@ def caption(args: argparse.Namespace):
         check_condition(len(args.checkpoints) == len(args.models),
                         "must provide checkpoints for each model")
 
+    # If the user has not explicitly set a batch size, set it based on the file type
+    if batch_size is None:
```

Comment:
I think you are missing a `batch_size = args.batch_size` here.
It might also make sense to factor this out into a function (def `set_batch_size(args)`)to not replicate code in captioner.py and translate.py.